python/utils/transactions_processor.py

```diff
@@ -152,3 +153,23 @@ def insert_to_db(self, parsed_objs, txn_version) -> None:
                         next_version=txn_version + 1,
                     )
                 )
+
+
+@event.listens_for(Base.metadata, "before_create")
+def create_schemas(target, connection, **kw):
+    schemas = set()
+    for table in target.tables.values():
+        if table.schema is not None:
+            schemas.add(table.schema)
+    for schema in schemas:
+        connection.execute(DDL("CREATE SCHEMA IF NOT EXISTS %s" % schema))
+
+
+@event.listens_for(Base.metadata, "after_drop")
+def drop_schemas(target, connection, **kw):
+    schemas = set()
+    for table in target.tables.values():
+        if table.schema is not None:
+            schemas.add(table.schema)
+    for schema in schemas:
+        connection.execute(DDL("DROP SCHEMA IF EXISTS %s" % schema))
```

Comment:
I think this will drop the entire schema if you delete a single table right now, which is not what we want.

Reply:
This event listener listens for drop on `Base.metadata`, so it will only drop the schema if the whole DB is deleted with `MetaData.drop_all`